gensio: Build error

Maintainer: @WereCatf (gensio), @nxhack (swig)
Environment: armvirt-32, snapshot sdk 2023-03-05

Description:
I am seeing this error when trying to build gensio:

```
make[8]: Entering directory '/media/jeff/Jekyll/Downloads/openwrt/testing/armvirt-32/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/gensio-2.4.2/c++/swig/pygensio'
/media/jeff/Jekyll/Downloads/openwrt/testing/armvirt-32/staging_dir/hostpkg/bin/swig -DHAVE_CONFIG_H -Wall -python -py3 \
        -o pygensio_wrap.cc -c++ \
        -I../../../c++/swig/pygensio -I../../../c++/include \
        -I../../../include \
        -I../../../include -I../../../c++/swig/include \
        ../../../c++/swig/pygensio/pygensio.i
Deprecated command line option: -py3. Ignored, this option is no longer supported.
SWIG:19: Error: PCRE regex matching is not available in this SWIG build.
make[8]: *** [Makefile:852: pygensio_wrap.cc] Error 1
make[8]: Leaving directory '/media/jeff/Jekyll/Downloads/openwrt/testing/armvirt-32/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/gensio-2.4.2/c++/swig/pygensio'
make[7]: *** [Makefile:569: all-recursive] Error 1
make[7]: Leaving directory '/media/jeff/Jekyll/Downloads/openwrt/testing/armvirt-32/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/gensio-2.4.2/c++/swig/pygensio'
make[6]: *** [Makefile:427: all-recursive] Error 1
make[6]: Leaving directory '/media/jeff/Jekyll/Downloads/openwrt/testing/armvirt-32/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/gensio-2.4.2/c++/swig'
make[5]: *** [Makefile:427: all-recursive] Error 1
make[5]: Leaving directory '/media/jeff/Jekyll/Downloads/openwrt/testing/armvirt-32/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/gensio-2.4.2/c++'
make[4]: *** [Makefile:484: all-recursive] Error 1
make[4]: Leaving directory '/media/jeff/Jekyll/Downloads/openwrt/testing/armvirt-32/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/gensio-2.4.2'
make[3]: *** [Makefile:416: all] Error 2
make[3]: Leaving directory '/media/jeff/Jekyll/Downloads/openwrt/testing/armvirt-32/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/gensio-2.4.2'
make[2]: *** [Makefile:207: /media/jeff/Jekyll/Downloads/openwrt/testing/armvirt-32/build_dir/target-arm_cortex-a15+neon-vfpv4_musl_eabi/gensio-2.4.2/.built] Error 2
make[2]: Leaving directory '/home/jeff/Code/personal/openwrt/packages/master/net/gensio'
time: package/feeds/packages/gensio/compile#0.71#0.35#1.88
    ERROR: package/feeds/packages/gensio failed to build.
make[1]: *** [package/Makefile:116: package/feeds/packages/gensio/compile] Error 1
make[1]: Leaving directory '/media/jeff/Jekyll/Downloads/openwrt/testing/armvirt-32'
make: *** [/media/jeff/Jekyll/Downloads/openwrt/testing/armvirt-32/include/toplevel.mk:224: package/gensio/compile] Error 2
```

The error message (`PCRE regex matching is not available in this SWIG build`) seems to indicate an issue with swig, though I have two questions:

* swig was configured to build without pcre in 2015 (https://github.com/openwrt/packages/commit/010cb4848d4014d745429db620ab237baf9da8f0) - is this still relevant now?
* gensio appears to build without error on buildbots. It appears that buildbots now have swig installed (https://github.com/openwrt/openwrt/pull/11666#issuecomment-1368998351, https://gitlab.com/openwrt/buildbot/-/commit/3776a9157532e69478ef03236d22254267a4f12e). Is gensio using the swig from the build host instead of the host package here? Is a swig package here still necessary?

--- a/libs/pcre2/Makefile
+++ b/libs/pcre2/Makefile
@@ -9,7 +9,7 @@ include $(TOPDIR)/rules.mk
 
 PKG_NAME:=pcre2
 PKG_VERSION:=10.37
-PKG_RELEASE:=$(AUTORELEASE)
+PKG_RELEASE:=2
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.bz2
 PKG_SOURCE_URL:=@SF/pcre/$(PKG_NAME)/$(PKG_VERSION)
@@ -28,6 +28,7 @@ PKG_CONFIG_DEPENDS:=\
 PKG_BUILD_DEPENDS:=zlib
 
 include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/host-build.mk
 include $(INCLUDE_DIR)/cmake.mk
 
 define Package/libpcre2/default
@@ -54,6 +55,18 @@ define Package/libpcre2-32
   $(call Package/libpcre2/default)
   TITLE:=A Perl Compatible Regular Expression library (32bit support)
 endef
+
+CMAKE_HOST_OPTIONS += \
+	-DBUILD_SHARED_LIBS=ON \
+	-DPCRE2_BUILD_PCRE2_8=ON \
+	-DPCRE2_BUILD_PCRE2_16=ON \
+	-DPCRE2_BUILD_PCRE2_32=ON \
+	-DPCRE2_DEBUG=OFF \
+	-DPCRE2_DISABLE_PERCENT_ZT=ON \
+	-DPCRE2_SUPPORT_JIT=OFF \
+	-DPCRE2_SHOW_REPORT=OFF \
+	-DPCRE2_BUILD_PCRE2GREP=OFF \
+	-DPCRE2_BUILD_TESTS=OFF
 
 CMAKE_OPTIONS += \
 	-DBUILD_SHARED_LIBS=ON \
@@ -92,3 +105,4 @@ endef
 $(eval $(call BuildPackage,libpcre2))
 $(eval $(call BuildPackage,libpcre2-16))
 $(eval $(call BuildPackage,libpcre2-32))
+$(eval $(call HostBuild))

--- a/net/gensio/Makefile
+++ b/net/gensio/Makefile
@@ -33,7 +33,7 @@ PKG_CONFIG_DEPENDS:= \
 	CONFIG_GENSIO_TCL \
 	CONFIG_GENSIO_SSHD
 
-PKG_BUILD_DEPENDS:=PACKAGE_python3-gensio:swig
+PKG_BUILD_DEPENDS:=PACKAGE_python3-gensio:swig/host
 
 include $(INCLUDE_DIR)/package.mk
 include ../../lang/python/python3-package.mk
@@ -132,7 +132,7 @@ $(call Package/gensio/Default)
   TITLE+= (Python3-bindings)
   SECTION:=lang
   CATEGORY:=Languages
-  DEPENDS:=+PACKAGE_python3-gensio:python3-light +libgensio
+  DEPENDS:=+PACKAGE_python3-gensio:python3-light +libgensiocpp
 endef
 
 define Package/python3-gensio/description

--- a/net/gensio/Makefile
+++ b/net/gensio/Makefile
@@ -129,9 +129,10 @@ endef
 
 define Package/python3-gensio
 $(call Package/gensio/Default)
-  TITLE+= (Python3-bindings)
+  TITLE+= (Python bindings)
   SECTION:=lang
   CATEGORY:=Languages
+  SUBMENU:=Python
   DEPENDS:=+PACKAGE_python3-gensio:python3-light +libgensiocpp
 endef
 

--- a/utils/swig/Makefile
+++ b/utils/swig/Makefile
@@ -20,6 +20,7 @@ PKG_LICENSE_FILES:=LICENSE
 
 PKG_HOST_ONLY:=1
 HOST_BUILD_PARALLEL:=1
+HOST_BUILD_DEPENDS:=pcre2/host
 
 include $(INCLUDE_DIR)/host-build.mk
 include $(INCLUDE_DIR)/package.mk
@@ -33,7 +34,7 @@ define Package/swig
 endef
 
 HOST_CONFIGURE_ARGS += \
-	--without-pcre
+	--with-pcre
 
 define Package/swig/description
   tool that generates bindings for various languages

